dbt/adapters/bigquery/impl.py

```diff
@@ -756,20 +756,22 @@ def get_table_options(
     ) -> Dict[str, Any]:
         opts = self.get_common_options(config, node, temporary)
 
-        if temporary:
-            expiration = 'TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL 12 hour)'
-            opts['expiration_timestamp'] = expiration
-
         if config.get('kms_key_name') is not None:
             opts['kms_key_name'] = "'{}'".format(config.get('kms_key_name'))
 
-        if config.get('require_partition_filter'):
-            opts['require_partition_filter'] = config.get(
-                'require_partition_filter')
-
-        if config.get('partition_expiration_days') is not None:
-            opts['partition_expiration_days'] = config.get(
-                'partition_expiration_days')
+        if temporary:
```

Comment:
Do we need all of this logic here, vs. the simpler logic proposed by @prratek in https://github.com/dbt-labs/dbt-core/pull/3386:
```python
        if config.get('require_partition_filter') and not temporary:
            opts['require_partition_filter'] = config.get(
                'require_partition_filter')
```

Reply:
It is ok with me. The reason why I made the nested conditions under `if temporary:` is the two conditions are a set for a temporary table. I will unnest the two.